### DISCUSSION TITLE
ISO Install Failure - MySQL?

### DISCUSSION THREAD

**Discussion by Sazerak** [Security-Onion-Solutions/securityonion#2382] [Unsupported Versions] [answered]
I've been trying to install SO using the ISO on my ESXi 6.7 (HP DL380 Gen7) and am constantly getting errors. As it stands, the following modules are in an error state:
- so-elastalert
- so-fleet
- so-playbook
- so-strelka-backend
- so-strelka-filestream
- so-strelka-frontend
- so-strelka-manager

The setup log shows the following errors, mainly related to MySQL:
[ERROR   ] Failure!
[ERROR   ] {'stdout': 'until nc -z 10.7.7.184 3306; do sleep 1; done : Timed out after 600 seconds', 'stderr': '', 'retcode': 1, 'pid': 20491}
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error: Unable to fetch current server version. Last error was: "MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server"
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error: Unable to fetch current server version. Last error was: "MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server"
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] {'stdout': 'until nc -z 10.7.7.184 3200; do sleep 1; done : Timed out after 30 seconds', 'stderr': '', 'retcode': 1, 'pid': 27038}
[ERROR   ] MySQL Error 2003: Can't connect to MySQL server on '10.7.7.184' (113)
[ERROR   ] MySQL Error 2003: Can't connect to MySQL server on '10.7.7.184' (113)
[ERROR   ] Command '['docker', 'exec', 'so-fleet', 'fleetctl', 'get', 'enroll-secret', 'default']' failed with return code: 1
[ERROR   ] stdout: Error response from daemon: Container 686323bd913154021d89a18d9154e96b4ab9d6f571022e6ece4a5bae53034fe7 is not running
[ERROR   ] retcode: 1
[ERROR   ] Command 'docker exec so-fleet fleetctl get enroll-secret default' failed with return code: 1
[ERROR   ] output: Error response from daemon: Container 686323bd913154021d89a18d9154e96b4ab9d6f571022e6ece4a5bae53034fe7 is not running
[ERROR   ] Command '['systemd-run', '--scope', 'systemctl', 'start', 'wazuh-agent.service']' failed with return code: 1
[ERROR   ] stderr: Running scope as unit run-5570.scope.
[ERROR   ] retcode: 1
[ERROR   ] Running scope as unit run-5570.scope.
[ERROR   ] Command '['systemd-run', '--scope', 'systemctl', 'start', 'wazuh-agent.service']' failed with return code: 1
[ERROR   ] stderr: Running scope as unit run-17732.scope.
[ERROR   ] retcode: 1
[ERROR   ] Running scope as unit run-17732.scope.
[ERROR   ] MySQL Error 2003: Can't connect to MySQL server on '10.7.7.184' (113)
[ERROR   ] MySQL Error 2003: Can't connect to MySQL server on '10.7.7.184' (113)
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error: Unable to fetch current server version. Last error was: "MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server"
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server
[ERROR   ] MySQL Error 1130: Host '10.7.7.99' is not allowed to connect to this MySQL server

Not sure what other information could be useful to post, but I've tried reverting the snapshot a few times and reinstalling. It also ends up pretty well the same (although I feel like sometimes the errors aren't completely constant).

Any help is appreciated.

**Comment by defensivedepth** [comment, marked as answer]
If you still have access to the failed install, can you please post the output of the following? The error messages show two different mysql IPs....

`sudo salt-call pillar.get global`

and

`sudo salt-call grains.get ip_interfaces`

**Reply by Sazerak** [reply]
local:
    ----------
    airgap:
        False
    dockernet:
        172.17.0.0
    fleet_custom_hostname:
        None
    fleet_hostname:
        N/A
    fleet_ip:
        N/A
    fleet_manager:
        False
    fleet_node:
        False
    fleet_packages-timestamp:
        N/A
    fleet_packages-version:
        1
    hnmanager:
        10.0.0.0/8
    ids:
        Suricata
    imagerepo:
        securityonion
    managerip:
        10.7.7.99
    managerupdate:
        1
    mdengine:
        SURICATA
    ntpserver:
    pipeline:
        redis
    proxy:
    sensoronikey:
        tJED9NqGHKSRLTKfgHz6
    soversion:
        2.3.10
    url_base:
        10.7.7.99
    wazuh:
        1


_____________________________________________________________


local:
    ----------
    bond0:
    docker0:
        - 172.17.0.1
    ens192:
        - 10.7.7.99
    ens34:
    lo:
        - 127.0.0.1
    veth3320619:


I ran these commands while simultaneously running the network configuration again, so I'm not sure if it will have affected the results, but this is what I'm getting. ens192 is my management interface (10.7.7.99) and ens34 is the monitoring interface (had the IP 10.7.7.184, but shows no ip after configuration. I think this is normal from what I read).

**Reply by defensivedepth** [reply]
@Sazerak I believe this is the same issue: https://github.com/Security-Onion-Solutions/securityonion/discussions/1896#discussioncomment-144262

`This is an unsupported configuration, which is why the install failed - The interface chosen as the Manager interface needs to be the default route for the system. We are adding a pre-flight check for this.`

**Reply by Sazerak** [reply]
Ah, ok. I'll have to look into that in the morning. This is the first time I'm playing around with the network settings on ESXi, so it's likely I'm guessing it's probably something in there that I'll have to figure out. Thanks!

**Reply by Sazerak** [reply]
Hmm.. So I didn't change anything on my ESXi, but the latest network configuration finished (not a fresh install, just ran the network config again) and now every module was showing OK other than MySQL which was showing as Missing. I ran sudo so-start mysql, and it started no problem. Now I can access my interface using .99 (although it used to tell me to use .184). My latest so-status is showing so-strelka-backend in error state, but all the others seem fine for now. Here are the outputs of the 2 commands from earlier again. Not sure if the configuration figured itself out, or if this is just a fluke:

local:
    ----------
    airgap:
        False
    dockernet:
        172.17.0.0
    fleet_custom_hostname:
        None
    fleet_hostname:
        SazSecOnion
    fleet_ip:
        10.7.7.99
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2020-12-17-20:16
    fleet_packages-version:
        2
    hnmanager:
        10.0.0.0/8
    ids:
        Suricata
    imagerepo:
        securityonion
    managerip:
        10.7.7.99
    managerupdate:
        1
    mdengine:
        SURICATA
    ntpserver:
    pipeline:
        redis
    proxy:
    sensoronikey:
        tJED9NqGHKSRLTKfgHz6
    soversion:
        2.3.10
    url_base:
        10.7.7.99
    wazuh:
        1



____________________________________________________________________________


local:
    ----------
    bond0:
    docker0:
        - 172.17.0.1
    ens192:
        - 10.7.7.99
    ens34:
    lo:
        - 127.0.0.1
    veth057de52:
    veth0b4d541:
    veth0f6aed7:
    veth22d0b88:
    veth2782768:
    veth2a67c0f:
    veth3320619:
    veth4ac8baf:
    veth4ba8b92:
    veth50d5b07:
    veth5153956:
    veth6ac074a:
    veth73a042c:
    veth73ede47:
    veth7b4873c:
    veth8943139:
    vetha60d3e9:
    vetha6b04be:
    vethb7c25a0:
    vethbce6f69:
    vethc109b18:
    vethd6369e8:
    vethe2a9408:
    vethe614ae5:
    vethf2ad228:


and I have no idea why I'm seeing all this on the 2nd commands... gremlins probably.

**Reply by defensivedepth** [reply]
Ya so the monitoring IP/interface was the default route for the system. Then when setup removed that IP during the first failed install, it no longer was the default route for the system and so the setup succeeded the 2nd time.

So next time, before you start setup, confirm that the monitoring interface doesnt have an IP. As I mentioned, our next release will have a check for this before setup runs.

**Reply by Sazerak** [reply]
Good to know. Should I be concerned about that 1 module still in error state (so-strelka-backend)?

I think what I'll do now is save this snapshot but revert to my pre-install snapshot and try the installer again making sure to remove the IP address from the monitoring interface. If I see this works well, then I'll revert again and try installing with The Hive. My original issue (before this post) was that the installer was hanging at 80% and it was caused by something related to The Hive. I decided to put that issue on hold though when this issue came up, but with any luck I should be good now and I can try The Hive again. Thanks for your help!

**Reply by Sazerak** [reply]
@defensivedepth Hoping I could pick your brain about my latest install attempt. I ended up modifying my network settings in CentOS by creating/modifying the files in /etc/sysconfig/network-scripts and that seemed to work to get the Management Interface as the default route and eliminate the IP address on the Monitoring Interface (modifying up ip or ifconfig commands didn't seem to stick). The install went well overall (Standalone install with everything except The Hive as I'm still having issues with it hanging). I did get 1 error, but reading the log, I feel like it might be normal to get this:

67% - APPLYING ZEEK SALT STATE
----
[INFO    ] Loading fresh modules for state activity
[INFO    ] Fetching file from saltenv 'base', ** done ** 'zeek/init.sls'
[INFO    ] Loading fresh modules for state activity
[INFO    ] Running state [zeek_state_not_allowed] at time 15:45:26.841874
[INFO    ] Executing state test.fail_without_changes for [zeek_state_not_allowed]
[ERROR   ] Failure!
[INFO    ] Completed state [zeek_state_not_allowed] at time 15:45:26.844648 (duration_in_ms=2.775)
local:
----------
          ID: zeek_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 15:45:26.841873
    Duration: 2.775 ms
     Changes:   

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.775 ms

Is this a normal error? It sounds like it is, but I would expect that if it is, I wouldn't have been warned about errors at the end of the installation. If it isn't normal, what could the cause be and will it have an impact on my experience?

After the installation, I once again noticed that so-mysql was missing. Running sudo so-start mysql solved that, but I'm not sure if I should be doing that or not. After rebooting, so-mysql still seems ok (I didn't try rebooting before restart the service, so not sure if just a reboot would have done the trick).

About 2-3 minutes later, I ran so-status again and once again, so-strelka-backend is in error state. After rebooting and waiting 5-10 minutes, so-strelka-backend didn't go into error state, so that's good.

Last thing I noticed is that although the configuration wizard runs through the option for so-allow, it never shows up at the end and I have to run so-allow manually to access the interface. I just thought I'd mention it if ever you guys want to troubleshoot that for your next release.

Overall, much better install that the first many many times, lol. I think I'll save this snapshot as my useable copy but still revert and try reinstalling with The Hive now that the installation is (mostly) perfect.

Thanks again for your help!

**Reply by defensivedepth** [reply]
So that Zeek error is a known bug that will be fix in the next release - it's because you selected Suricata for metadata instead of Zeek. The strelka issue is also a known issue that will be fixed in the next release :)   It's not actually an error, it's just displaying it as such.

**Reply by Sazerak** [reply]
Good to know, thanks!

**Reply by Sazerak** [reply]
I just wanted to post one last update for the benefit of anyone else reading this that has similar issues. The network fix I applied earlier seems to have weeded out pretty well all of my issues. Even installing with The Hive works now, it doesn't hang anymore (although I can't say for sure if that's due to the network fix or not).

It occurred to me that since the configuration finishes with an error, it doesn't prompt me to reboot before using SO, which probably explains the missing so-mysql and maybe the erroneous so-strelka-backend. After rebooting, everything seems fine with the exception of having to run so-allow in order to access the interface, even though it should have ran during the config (interestingly enough, the sosetup.log doesn't indicate that so-allow actually runs).

Thank you again to @defensivedepth for all his help! I feel like just going through the installation/configuration process has helped me learn a lot and I can't wait to actually start using and learning SO.

**Reply by defensivedepth** [reply]
@Sazerak No problem! Glad we were able to get it all sorted out. Ya, the networking issue would have messed up a bunch of stuff during the install.

If you haven't seen it yet, make sure to take a look at our new FREE training that should give you a good start on digging into and learning the platform: Security Onion Essentias - https://www.youtube.com/playlist?list=PLljFlTO9rB155aYBjHw2InKkSMLuhWpxH

**Reply by Sazerak** [reply]
Just wanted to add one more thing for anyone still tracking this discussion. I had brought up the fact that so-allow wasn't being run even though I had chosen it in the config. Today, I tried reinstalling SO using ZEEK rather than SURICATA. This time, the config went to the end (did not stop at 95% due to the zeek "ERROR"), and so-allow ran at 97%. So that was the cause of it. Once the dev team fixes the known zeek error when using suricata, it should be good. In the meantime, anyone installing with suricata simply needs to run so-allow post-installation. Hope this helps!

**Reply by dougburks** [reply]
The Suricata metadata error was resolved in version 2.3.21:
https://docs.securityonion.net/en/2.3/release-notes.html#changes
